modules/iap/iap.tf

```diff
@@ -19,145 +19,72 @@ resource "random_string" "random" {
   upper   = false
 }
 
-# TODO refactor Jupyter and Frontend to be one
-# Jupyter IAP
-resource "google_compute_global_address" "jupyter_ip_address" {
-  count        = var.jupyter_add_auth && var.jupyter_url_domain_addr == "" ? 1 : 0
+# IAP
+resource "google_compute_global_address" "ip_address" {
+  count        = var.add_auth && var.url_domain_addr == "" ? 1 : 0
   provider     = google-beta
   project      = var.project_id
-  name         = "jupyter-address-${random_string.random.result}"
-  address_type = "EXTERNAL"
-  ip_version   = "IPV4"
-}
-
-# Helm Chart IAP
-resource "helm_release" "iap_jupyter" {
-  count            = var.jupyter_add_auth ? 1 : 0
-  name             = "iap-jupyter"
-  chart            = "${path.module}/charts/iap/"
-  namespace        = var.namespace
-  create_namespace = true 
-  # timeout increased to support autopilot scaling resources, and give enough time to complete the deployment 
-  timeout = 1200
-  set {
-    name  = "iap.backendConfig.name"
-    value = var.jupyter_k8s_backend_config_name
-  }
-
-  set {
-    name  = "iap.secret.name"
-    value = var.jupyter_k8s_iap_secret_name
-  }
-
-  set {
-    name  = "iap.secret.client_id"
-    value = base64encode(var.jupyter_client_id)
-  }
-
-  set {
-    name  = "iap.secret.client_secret"
-    value = base64encode(var.jupyter_client_secret)
-  }
-
-  set {
-    name  = "iap.managedCertificate.name"
-    value = var.jupyter_k8s_managed_cert_name
-  }
-
-  set {
-    name  = "iap.managedCertificate.domain"
-    value = var.jupyter_url_domain_addr != "" ? var.jupyter_url_domain_addr : "${google_compute_global_address.jupyter_ip_address[0].address}.nip.io"
-  }
-
-  set {
-    name  = "iap.ingress.staticIpName"
-    value = var.jupyter_url_domain_addr != "" ? var.jupyter_url_domain_name : "${google_compute_global_address.jupyter_ip_address[0].name}"
-  }
-
-  set {
-    name  = "iap.ingress.name"
-    value = var.jupyter_k8s_ingress_name
-  }
-
-  set {
-    name  = "iap.ingress.backendServiceName"
-    value = var.jupyter_k8s_backend_service_name
-  }
-
-  set {
-    name  = "iap.ingress.backendServicePort"
-    value = var.jupyter_k8s_backend_service_port
-  }
-}
-
-# TODO set the member allowlist
-
-# Frontend IAP
-resource "google_compute_global_address" "frontend_ip_address" {
-  count        = var.frontend_add_auth && var.frontend_url_domain_addr == "" ? 1 : 0
-  provider     = google-beta
-  project      = var.project_id
-  name         = "frontend-address-${random_string.random.result}"
+  name         = "${var.app_name}-address-${random_string.random.result}"
   address_type = "EXTERNAL"
   ip_version   = "IPV4"
 }
 
 # Helm Chart IAP
 resource "helm_release" "iap_frontend" {
-  count            = var.frontend_add_auth ? 1 : 0
-  name             = "iap-frontend"
+  count            = var.add_auth ? 1 : 0
```

Comment:
add_auth is not required here.. module will always be called when add_auth is true